scheduler/scheduler.go

```diff
@@ -17,12 +18,15 @@ func New(config config.Configuration) *Scheduler {
 }
 
 func (scheduler *Scheduler) GetBackend() (string, error){
-	numberOfServers := len(scheduler.Servers)
+	var available []string
+	available, _ = ping.Healthcheck(scheduler)
```

Comment:
Eventually, periodic health checks should be done instead of performing a health check whenever a request arrives.